printer/printer_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package printer
+
+import (
+	"bufio"
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockWriter struct {
+	buffer []byte
+}
+
+func (w *mockWriter) Write(b []byte) (n int, err error) {
+	w.buffer = append(w.buffer, b...)
+	return len(w.buffer) - len(b), nil
+}
+
+func TestPrintT(t *testing.T) {
+	w := bufio.NewWriter(&bytes.Buffer{})
+	printer.writer = w
+	printer.Format = FormatPlain
+
+	ts := struct {
+		ID int
+	}{
+		ID: 123,
+	}
+
+	t.Run("should execute template", func(t *testing.T) {
+		tpl := `testing template {{.ID}}`
+		PrintT(tpl, ts)
+		assert.Equal(t, "testing template 123", printer.Lines[0], "should be equal")
```

Comment:
The `assert.Equal` explains the inequality in case of error, so the message gets redundant:

```suggestion
		assert.Equal(t, "testing template 123", printer.Lines[0])
```